### PR TITLE
[ticket/14051] Replace sql array of viewforum sort topics function

### PR DESCRIPTION
**Pull request by RMcGirr83**
https://tracker.phpbb.com/browse/PHPBB3-14051

PHPBB3-14051

### CODE REVIEW

**Comment by RMcGirr83** [comment]
https://github.com/phpbb/phpbb/pull/379

**Comment by RMcGirr83** [comment]
!set event

**Comment by RMcGirr83** [comment]
!set Event

File: phpBB/viewforum.php [hunk 1]
```diff
@@ -274,16 +274,41 @@
...
+			AND ' . $phpbb_content_visibility->get_visibility_sql('topic', $forum_id),
+	);
+
+	/**
+	* Modify the sort data SQL query for getting additional fields if needed
+	*
+	* @event core.viewforum_modify_sort_data_sql
+	* @var int		forum_id		The forum_id whose topics are being listed
+	* @var int		start			Variable containing start for pagination
+	* @var	array	sort_days		The oldest topic displayable in elapsed days
```
Comment (marc1706) [inline phpBB/viewforum.php]: Isn't this an integer?

File: phpBB/viewforum.php [hunk 2]
```diff
@@ -274,16 +274,41 @@
...
+	* @var	array	sort_key		The sorting by. It is one of the first character of (in low case):
```
Comment (marc1706) [inline phpBB/viewforum.php]: string

File: phpBB/viewforum.php [hunk 3]
```diff
@@ -274,16 +274,41 @@
...
+	*								Author, Post time, Replies, Subject, Views
+	* @var	array	sort_dir		Either "a" for ascending or "d" for descending
```
Comment (marc1706) [inline phpBB/viewforum.php]: string

File: phpBB/viewforum.php [hunk 4]
```diff
@@ -274,16 +274,41 @@
...
+	$sql_array = array(
+		'SELECT'	=> 'COUNT(t.topic_id) AS num_topics',
+		'FROM'		=> array(
+			TOPICS_TABLE	=> 't',
+		),
+		'WHERE'		=> 't.forum_id = ' . $forum_id . '
+			AND (t.topic_last_post_time >= ' . $min_post_time . '
+				OR t.topic_type = ' . POST_ANNOUNCE . '
+				OR t.topic_type = ' . POST_GLOBAL . ')
+			AND ' . $phpbb_content_visibility->get_visibility_sql('topic', $forum_id),
```
Comment (gooof) [inline phpBB/viewforum.php]: ('topic', $forum_id, 't.')

**Comment by RMcGirr83** [comment]
@marc1706 may want to fix ~lines 424 to 429 for types on properties in the event.  New Ticket or?

**Pull request merged**